### PR TITLE
Draft: dts: apq8074: Samsung Galaxy Tab PRO 8.4 WIFI (SM-T320) (mondrianwifi)

### PR DESCRIPTION
**Pull request by exkc**

### CODE REVIEW

**Comment by exkc** [comment]
decive tree for it : https://github.com/LineageOS/android_kernel_samsung_klte/blob/61c96565d20a3ff234a8f1e57e1b21f99b079ea0/arch/arm/boot/dts/msm8974/msm8974-sec-mondrian-r12.dtsi

File: dts/msm8974/rules.mk [hunk 1]
```diff
@@ -6,4 +6,5 @@ DTBS += \
 	$(LOCAL_DIR)/msm8974-samsung-r10.dtb \
 	$(LOCAL_DIR)/msm8974-samsung-r14.dtb \
 	$(LOCAL_DIR)/msm8974pro-ab-pm8941-mtp.dtb \
+	$(LOCAL_DIR)/apq8074-samsung-mondrianwifi.dtb \
```
Comment (TravMurav) [inline dts/msm8974/rules.mk:9]: Please follow the naming scheme alike to other devces

File: dts/msm8974/apq8074-samsung-mondrianwifi.dts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#include <lk2nd.h>
+
+/ {
+	qcom,msm-id = <0xb801ff01 0x0c 0x00>;
+	compatible = "qcom,apq8074-cdp", "qcom,apq8074", "lk2nd,device";
+
+		model = "mondrianwifi";
+		compatible = "mondrianwifi", "qcom,apq8074", "lk2nd,device";
+
+
```
Comment (TravMurav) [inline dts/msm8974/apq8074-samsung-mondrianwifi.dts:15]: Sorry, but this looks nonsensical to me. Please look at the nearby samsung files for examples.

**Comment by exkc** [comment]
lk2nd seem to not booting on it.

File: dts/msm8974/apq8074-samsung-mondrianwifi.dts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	qcom,msm-id = <0xB801FF01 0 0>,
+	 <0xB801FF01 7 0>,
+	 <0xB801FF01 8 0>,
+	 <0xB801FF01 12 0>;
+
+		model = "mondrianwifi";
+		compatible = "mondrianwifi", "qcom,apq8074", "lk2nd,device";
+
+
+};
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8974/apq8074-samsung-mondrianwifi.dts:18]:
```suggestion
};
```

File: dts/msm8974/apq8074-samsung-mondrianwifi.dts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#include <lk2nd.h>
+
+/ {
+	qcom,msm-id = <0xB801FF01 0 0>,
+	 <0xB801FF01 7 0>,
+	 <0xB801FF01 8 0>,
+	 <0xB801FF01 12 0>;
+
+		model = "mondrianwifi";
+		compatible = "mondrianwifi", "qcom,apq8074", "lk2nd,device";
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8974/apq8074-samsung-mondrianwifi.dts:15]:
```suggestion
		compatible = "samsung,mondrianwifi", "qcom,apq8074", "lk2nd,device";
```

File: dts/msm8974/apq8074-samsung-mondrianwifi.dts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	qcom,msm-id = <0xB801FF01 0 0>,
+	 <0xB801FF01 7 0>,
+	 <0xB801FF01 8 0>,
+	 <0xB801FF01 12 0>;
+
+		model = "mondrianwifi";
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8974/apq8074-samsung-mondrianwifi.dts:14]:
```suggestion
		model = "Samsung Galaxy Tab PRO 8.4 (SM-T320)";
```

**Review by wonderfulShrineMaidenOfParadise — COMMENTED** [review]
> dts: msm8916: Samsung Galaxy Tab PRO 8.4 (SM-T320) (mondrianwifi)

It's not msm8916, is it?

File: dts/msm8974/apq8074-samsung-mondrianwifi.dts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+#include <lk2nd.h>
+
+/ {
+	
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <0xB801FF01 0 0>,
+	 <0xB801FF01 7 0>,
+	 <0xB801FF01 8 0>,
+	 <0xB801FF01 12 0>;
+	mondrianwifi {
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8974/apq8074-samsung-mondrianwifi.dts:15]:
```suggestion
	 <0xB801FF01 12 0>;

	mondrianwifi {
```

A newline between last property and first node.

File: dts/msm8974/apq8074-samsung-mondrianwifi.dts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+		compatible = "samsung,mondrianwifi", "qcom,apq8074", "lk2nd,device";
+		model = "Samsung Galaxy Tab PRO 8.4 WIFI (SM-T320)";
+		lk2nd,match-cmdline = "*samsung.hardware=SM-T320*";
+		lk2nd,smd-rpm-hack-opening;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8974/apq8074-samsung-mondrianwifi.dts:19]: Originally this is required on some msm8916/msm8939 samsung devices to boot a kernel. Make sure that you need it, especially when the platform is apq8074 instead.

Comment (exkc) [reply dts/msm8974/apq8074-samsung-mondrianwifi.dts:19]: > Make sure that you need it, especially when the platform is apq8074 instead.

Well even with this,  it cant boot to lk2nd.

Comment (wonderfulShrineMaidenOfParadise) [reply dts/msm8974/apq8074-samsung-mondrianwifi.dts:19]: https://github.com/msm8953-mainline/linux/commit/0a4b7f032a6437e812565da35c6e2a4a6cb2a2c2

See this for the details, if you are interested.

Comment (exkc) [reply dts/msm8974/apq8074-samsung-mondrianwifi.dts:19]: humm the hack look like it is only gnooa affect booting that kernal so it has nothing to do with my issue (It cant boot into lk2nd).

So maybe my issue is caused by android 4's bl?

Is there any there that i could use to fix this?

**Pull request closed without merging**